module/solutions/helper.js

```diff
@@ -1130,6 +1134,29 @@ module.exports = class SolutionsHelper {
 					} else {
 						filterQuery['scope.entityType'] = { $in: entityTypes }
 					}
+
+					let userRoleInfo = _.omit(data, ['filter', 'factors', 'role', 'type', 'tenantId', 'orgId'])
+
+					let tenantDetails = await userService.tenantDetails(origin)
+					if (
+						tenantDetails.data.meta.hasOwnProperty('factors') &&
+						tenantDetails.data.meta.factors.length > 0
+					) {
+						let factors = tenantDetails.data.meta.factors
+						let queryFilter = []
+
+						// Build query based on each key
+						factors.forEach((factor) => {
+							let scope = 'scope.' + factor
+							let values = userRoleInfo[factor]
+							if (!Array.isArray(values)) {
+								queryFilter.push({ [scope]: { $in: values.split(',') } })
+							} else {
+								queryFilter.push({ [scope]: { $in: [...values] } })
+							}
+						})
+						filterQuery['$and'] = queryFilter
```

Comment:
@MallanagoudaB if we want we can add a console for the query so that we can using it for server testing and later we can remove it